internal/config/ociartifact/impl.go

```diff
@@ -19,8 +20,11 @@ type Impl interface {
 	ManifestFromBlob([]byte, string) (manifest.Manifest, error)
 	ManifestConfigInfo(manifest.Manifest) types.BlobInfo
 	LayerInfos(manifest.Manifest) []manifest.LayerInfo
+	DatabaseRead(*diskv.Diskv, string) ([]byte, error)
+	DatabaseErase(*diskv.Diskv, string) error
 	GetBlob(context.Context, types.ImageSource, types.BlobInfo, types.BlobInfoCache) (io.ReadCloser, int64, error)
 	ReadAll(io.Reader) ([]byte, error)
+	DatabaseWrite(*diskv.Diskv, string, []byte) error
```

Comment:
Perhaps?

```suggestion
	DatabaseRead(*diskv.Diskv, string) ([]byte, error)
	DatabaseWrite(*diskv.Diskv, string, []byte) error
	DatabaseErase(*diskv.Diskv, string) error
	GetBlob(context.Context, types.ImageSource, types.BlobInfo, types.BlobInfoCache) (io.ReadCloser, int64, error)
	ReadAll(io.Reader) ([]byte, error)

```

Reply:
I am not a fan calling this "database" everywhere. It's a cache, really, and we also use it as such.

It might nice to move this to an interface, then embed it here? This would allow us to have more flexibility over what backend is the cache using–reading and writing through memory for testing? perhaps just a passthrough? 

Perhaps even ability to disable this via a configuration option?

Thoughts?

Reply:
Changed by not using diskv.